jxl/src/api/color.rs

```diff
@@ -1510,6 +1564,432 @@ fn create_table_curve(
     Ok(table)
 }
 
+// ============================================================================
+// HDR Tone Mapping Implementation
+// ============================================================================
+
+/// BT.2408 HDR to SDR tone mapper.
+/// Maps PQ content from source range (e.g., 0-10000 nits) to target range (e.g., 0-250 nits).
+struct Rec2408ToneMapper {
+    source_range: (f32, f32), // (min, max) in nits
+    target_range: (f32, f32),
+    luminances: [f32; 3], // RGB luminance coefficients (Y values)
+
+    // Precomputed values
+    pq_mastering_min: f32,
+    #[allow(dead_code)] // Stored for potential future use / debugging
+    pq_mastering_max: f32,
+    pq_mastering_range: f32,
+    inv_pq_mastering_range: f32,
+    min_lum: f32,
+    max_lum: f32,
+    ks: f32,
+    inv_one_minus_ks: f32,
+    normalizer: f32,
+    inv_target_peak: f32,
+}
+
+impl Rec2408ToneMapper {
+    fn new(source_range: (f32, f32), target_range: (f32, f32), luminances: [f32; 3]) -> Self {
+        let pq_mastering_min = Self::inv_eotf(source_range.0);
+        let pq_mastering_max = Self::inv_eotf(source_range.1);
+        let pq_mastering_range = pq_mastering_max - pq_mastering_min;
+        let inv_pq_mastering_range = 1.0 / pq_mastering_range;
+
+        let min_lum =
+            (Self::inv_eotf(target_range.0) - pq_mastering_min) * inv_pq_mastering_range;
+        let max_lum =
+            (Self::inv_eotf(target_range.1) - pq_mastering_min) * inv_pq_mastering_range;
+        let ks = 1.5 * max_lum - 0.5;
+
+        Self {
+            source_range,
+            target_range,
+            luminances,
+            pq_mastering_min,
+            pq_mastering_max,
+            pq_mastering_range,
+            inv_pq_mastering_range,
+            min_lum,
+            max_lum,
+            ks,
+            inv_one_minus_ks: 1.0 / (1.0 - ks).max(1e-6),
+            normalizer: source_range.1 / target_range.1,
+            inv_target_peak: 1.0 / target_range.1,
+        }
+    }
+
+    /// PQ EOTF inverse (Encoded from Display)
+    #[allow(clippy::excessive_precision)] // PQ constants are exact per SMPTE ST 2084
+    fn inv_eotf(luminance: f32) -> f32 {
+        // PQ transfer function constants from SMPTE ST 2084
+        const M1: f32 = 0.1593017578125;
+        const M2: f32 = 78.84375;
+        const C1: f32 = 0.8359375;
+        const C2: f32 = 18.8515625;
+        const C3: f32 = 18.6875;
+
+        let y = (luminance / 10000.0).max(0.0);
+        let ym1 = y.powf(M1);
+        ((C1 + C2 * ym1) / (1.0 + C3 * ym1)).powf(M2)
+    }
```

Comment:
I'm pretty sure PQ EOTF and HLG OOTF is already implemented in `color::tf`, maybe you can reuse those.

Reply:
Good catch! The existing `color::tf` functions work on slices with intensity_target scaling, while the ICC LUT needs per-pixel EOTF calls returning direct values. Could wrap them, but adds overhead for 729 LUT entries. Happy to refactor if you prefer consistency.